File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode curr, prev;
        curr = head;
        prev = head;
        // Edge Case:
        if(curr == null)
            return null;
        while(curr != null){
            if(prev.val == curr.val){
                curr = curr.next;
            }
            else{
                prev.next = curr;
                prev = curr;
            }
        }
        
        // Lastly Assigining the Prev to curr
        if(prev != null)
            prev.next = curr;
        
        return head;
    }
}

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let curr = head, prev = head;
    
    // Edge Case:
    if(curr == null)
        return null;
    while(curr){
        if(curr.val == prev.val)
            curr = curr.next;
        else{
            prev.next = curr;
            prev = curr;
        }
    }
    
    // For the last Case:
    if(prev != null){
        prev.next = curr;
    }
    
    // Return the Answer
    return head;
};